src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -64,43 +76,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
       response.getWriter().println("Invalid Query");
       return;
     }
-
     String query = encodeValue(param);
-    List<String> allLinks = new ArrayList<>();
-
-    // TODO: after implementing scraping, consider changing getLink calls to a
-    // for-loop
-
-    /*
-     * Send request to retrieve card content through w3Schools site link from Google
-     * CSE
-     */
-    String w3Link = getLink(W3_CSE_ID, query);
-    if (w3Link != null) {
-      allLinks.add(w3Link);
-      // TODO: Call scraping function to return JSON card content
-    }
-
-    /*
-     * Send request to retrieve card content through StackOverflow site link from
-     * Google CSE
-     */
-    String stackLink = getLink(STACK_CSE_ID, query);
-    if (stackLink != null) {
-      allLinks.add(stackLink);
-      // TODO: Call stackoverflow API to return JSON card content
-    }
-
-    /*
-     * Send request to retrieve card content through GeeksForGeeks site link from
-     * Google CSE
-     */
-    String geeksLink = getLink(GEEKS_CSE_ID, query);
-    if (geeksLink != null) {
-      allLinks.add(geeksLink);
-      // TODO: Call scraping function to return JSON card content
+    List<Card> allCards = new ArrayList<>();
+    for (Client client : clients) {
+      String link = getLink(client.getCseId(), query);
+      Card card = client.search(link);
+      if (card == null) {
+        break;
+      }
+      allCards.add(card);
     }
-
+    request.setAttribute("cardList", allCards);
```

Comment:
I think Aaron changed this to `request.setAttribute(CARD_LIST_LABEL, allCards)` in his #27